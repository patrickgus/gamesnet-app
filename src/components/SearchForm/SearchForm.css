.searchBar {
  display: flex;
  align-items: center;
  margin-top: 15px;
  width: 340px;
  color: rgb(0, 224, 224);
}

.searchBar label {
  margin-right: 5px;
}

.searchBar button {
  background-color: rgb(234, 0, 255);;
  border-color: rgb(194, 68, 194);
  border-radius: 4px;
  margin-left: 5px;
  font-weight: bold;
  cursor: pointer;
}

.searchBar button:hover {
  background-color: rgb(139, 0, 151);
}

#search {
  width: 114px;
  border: 1px solid rgb(19, 37, 85);
  border-radius: 4px;
  padding-left: 5px;
  margin-right: 5px;
}

#sort {
  border: 1px solid rgb(19, 37, 85);
  border-radius: 4px;
}

@media screen and (max-width: 350px) {
  #search {
    width: 80px;
  }
}

@media screen and (min-width: 680px) {
  .searchBar {
    margin: 15px auto 0;
    width: 480px;
  }

  #search {
    width: 250px;
    margin-right: 10px;
  }
}
