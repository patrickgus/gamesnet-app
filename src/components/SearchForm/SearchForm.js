import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import GameListContext from "../../contexts/GameListContext";
import "./SearchForm.css";

class SearchForm extends Component {
  state = {
    searchTerm: "",
    sort: "title"
  };

  static contextType = GameListContext;

  handleSearch = e => {
    this.setState({
      searchTerm: e.target.value
    });
  };

  handleSort = e => {
    this.setState({
      sort: e.target.value
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { gameList } = this.context;
    const { sort } = this.state;

    let filteredList = [];

    if (this.state.searchTerm !== "") {
      filteredList = gameList.filter(game => {
        const title = game.title.toLowerCase();
        const filter = this.state.searchTerm.toLowerCase();
        return title.includes(filter);
      });
    } else {
      filteredList = gameList;
    }

    if (sort === "title") {
      filteredList.sort((a, b) => {
        return a[sort] > b[sort] ? 1 : a[sort] < b[sort] ? -1 : 0;
      });
    } else if (sort === "avg_rating") {
      filteredList.sort((a, b) => {
        return b[sort] - a[sort];
      });
    }
    this.context.setGameList(filteredList);

    this.props.history.push("/games");
  };

  render() {
    return (
      <form className="searchBar" onSubmit={this.handleSubmit}>
        <label htmlFor="search">Search: </label>
        <input
          type="text"
          id="search"
          name="search"
          value={this.state.searchTerm}
          onChange={this.handleSearch}
        />

        <label htmlFor="sort">Sort: </label>
        <select
          id="sort"
          name="sort"
          value={this.state.sort}
          onChange={this.handleSort}
        >
          <option value="title">Title</option>
          <option value="avg_rating">Rating</option>
        </select>
        <button type="submit">Go</button>
      </form>
    );
  }
}

export default withRouter(SearchForm);
