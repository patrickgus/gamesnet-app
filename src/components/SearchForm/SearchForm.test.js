import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import SearchForm from "./SearchForm";

describe("SearchForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <SearchForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the form as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SearchForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
