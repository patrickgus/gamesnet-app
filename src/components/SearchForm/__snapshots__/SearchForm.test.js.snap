// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders the form as expected 1`] = `
<form
  className="searchBarNav"
  onSubmit={[Function]}
>
  <label
    htmlFor="search"
  >
    Search: 
  </label>
  <input
    id="search"
    name="search"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    htmlFor="sort"
  >
    Sort: 
  </label>
  <select
    id="sort"
    name="sort"
    onChange={[Function]}
    value="title"
  >
    <option
      value="title"
    >
      Title
    </option>
    <option
      value="avg_rating"
    >
      Rating
    </option>
  </select>
  <button
    type="submit"
  >
    Go
  </button>
</form>
`;
